university/src/university/Dorm.java

```diff
@@ -0,0 +1,37 @@
+package university;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Dorm implements Cloneable{
+    private int roomNumber;
+    private List<Integer> roomList;
+    public Dorm(){
+        roomList = new ArrayList<>();
+    }
+    public Dorm(List<Integer> rooms){
+        roomList = rooms;
+    }
+    public void loadRooms(){
+        roomList.addAll(List.of(101, 102, 103, 104, 105, 201, 202, 203, 204, 205, 301, 302, 303, 304, 305));
+    }
+
+    public List<Integer> getRoomList() {
+        return roomList;
+    }
+    @Override
+    public Object clone() throws CloneNotSupportedException{
+        List<Integer> tempList = new ArrayList<>();
+        for (Integer i: roomList) {
```

Comment:
if i am not mistaken, you could use small `int` in the `foreach` - the cast to reference types should be done automatically

Reply:
PS: if you are not expecting `null` values, a primitive type is generally better to be used